### ISSUE TITLE
Monitoring validation set accuracy during training

### ISSUE THREAD

**Issue by tambetm** [NervanaSystems/neon#56]
As I understand, currently metrics are only evaluated at the end of training session. Is there a way to monitor validation accuracy during training, so that I can cancel training session if results don't improve?

Similar question - are there any existing utilities to plot loss and accuracy over training session?

**Comment by scttl** [comment]
HI,

Yes with some python edits it should be possible to report additional metrics or other dataset partition performance at the end of each epoch.  We only report training set error by default to keep things running quickly but if you're ok with the additional overhead you should be able to create new metrics at the model level (either edit the yaml, or pass these into your model constructor call), then adjust your model's `fit()` function to:

1. generate predictions (see your model's `predict_fullset()` or `predict_generator()` functions)
2. use these predictions with your dataset partition's expected outputs to update your metric's statistics (see your metric(s) `add()` function)
3. report your metric's value (see your metric(s) `report()` function)

You can see a lot of this actually carried out in `fit_predict_err.py`'s `run()` method.

As for your second question regarding plotting, this is something we're actively developing but don't yet have the capability to visualize metric stats per epoch.  Most of the plotting currently in neon is based around timing/flops, and numeric representation.

**Comment by scttl** [comment]
Closing as we now support per epoch metric printing via #61
